### PR TITLE
ActionBar: Add new component for submenu

### PR DESCRIPTION
**Pull request by pksjce**
Closes #https://github.com/github/primer/issues/3132

This makes use of new capability in ActionMenu to have Submenus. We can use it to make sure that menus are supported with IconButton trigger in the ActionBar. The intent is to be able to convert it into a SubMenu when the overflow behaviour kicks in.

#### New

https://github.com/primer/react/assets/417268/cb59f7c8-f634-47b7-8a3a-0c2c0f094daa

### Rollout strategy

- [x] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Testing & Reviewing

Check out the new submenu story.

### Merge checklist

- [ ] Added/updated tests
- [x] Added/updated documentation
- [x] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: packages/react/src/drafts/ActionBar/ActionBar.tsx
Comment (siddharthkp) [inline packages/react/src/drafts/ActionBar/ActionBar.tsx:17]: Oops :)

**Review by siddharthkp — APPROVED** [review]
Other than linting errors, looks good.

What are your thoughts on the name: just `ActionBar.Menu` instead of `ActionBar.SubMenu`?

Approving in advance coz experimental component

**Comment by pksjce** [comment]
I'm still waiting for this PR to get merged for this one to be viable.https://github.com/primer/react/pull/4486

**Comment by broccolinisoup** [comment]
Removing myself from the reviewers for now, feel free to add back in when the manage to merge back the nested menus 🙌🏻

**Pull request closed without merging**
